Source/CarthageKit/Simulator.swift

```diff
@@ -45,8 +45,15 @@ internal func selectAvailableSimulator(of sdk: SDK, from data: Data) -> Simulato
 		return nil
 	}
 	let platformName = sdk.platform.rawValue
-	let allTargetSimulators = devices
-		.filter { $0.key.hasPrefix(platformName) }
+	func reducePlatformNames(_ result: inout [String: [Simulator]], _ entry: (key: String, value: [Simulator])) {
+		guard let range = entry.key.range(of: platformName) else { return }
+		let platform = entry.key
```

Comment:
This implementation is bit complex to read.

I suggest implementing a function to parse OS version from its identifiers.
And you should add tests for that function.

```swift
func parsePlatformVersion(from identifier: String) -> String?
```